airbyte-protocol/models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml

```diff
@@ -94,6 +98,45 @@ definitions:
       message:
         description: "the log message"
         type: string
+  AirbyteTraceMessage:
+    type: object
+    additionalProperties: true
+    required:
+      - type
+      - emitted_at
+    properties:
+      type:
+        description: "the type of trace message"
+        type: string
+        enum:
+          - ERROR
+      emitted_at:
+        description: "the time in ms that the message was emitted"
+        type: number
+      error:
+        description: "error data from the sync"
+        "$ref": "#/definitions/AirbyteErrorTraceMessage"
+  AirbyteErrorTraceMessage:
+    type: object
+    additionalProperties: true
+    required:
+      - message
+    properties:
+      message:
+        description: The human readable message that will be shown to the user in the UI
+        type: string
+      internal_message:
+        description: The less human readable, more technical message showing the reason for the failure, e.g. the first line of a stack trace
```

Comment:
I agree with you. Evan and I discussed this in the spec - @evantahler I think you were thinking that it's not necessarily going to be the first line of a stack trace and could be some other kind of message?

Reply:
Looking at the conversation again - @evantahler maybe you only wanted this field for AirbyteTraceMessages that are not errors? https://docs.google.com/document/d/1ctrj3Yh_GjtQ93aND-WH3ocqGxsmxyC3jfiarrF6NY0/edit?disco=AAAAYr7ZeRQ

Reply:
Maybe this is backwards thinking, but the inspiration once again comes from our (future) usage of these messages to create FailureReasons, and we've got all 3: `message`, `internal_message`, and `stack_trace`.  I do think there is value for both `internal_message`, and `stack_trace`, for a few reasons:

Reply:
Maybe this is backwards thinking, but the inspiration for the message once again comes from our (future) usage of these messages to create FailureReasons, and we've got all 3: `message`, `internal_message`, and `stack_trace` in FailureReasons.  

I do think there is value for both `internal_message`, and `stack_trace`.  An error like
```
message: API Key not accepted - check your credentials
internal_message: 403: Access Denied
stack_trace: "403: Access Denied" from connect() on line 101...
```
could come from a number of different places in the code (really, any time we make an HTTP request).  Analyzing these by internal message (which will always be the same) even though they come from different stack traces will be helpful.  Grouping by the full stack trace without doing some extraction won't be helpful.  The "message" might be decorated better to say "...check your username" vs "... check your apiKey" if we are able to do that. 

Also, I think the connector, using the same programming language that generated the stack trace, is the best place to "extract" the most meaningful message from the stack.

Reply:
Based on Evan's comment I think internal_message makes sense to keep, but I agree that we should remove `less human readable` from the description, and probably also remove `e.g. the first line of a stack trace`. Intead this could more simple like `The message of the internal error that caused the failure` or something along those lines